src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -2423,8 +2443,9 @@ private SyntaxToken LexDirectiveToken()
             TokenInfo info = default(TokenInfo);
             this.ScanDirectiveToken(ref info);
             var errors = this.GetErrors(leadingTriviaWidth: 0);
-            var trailing = this.LexDirectiveTrailingTrivia(info.Kind == SyntaxKind.EndOfDirectiveToken);
-            return Create(in info, null, trailing, errors);
+            var directiveTriviaCache = GetDirectiveTriviaCache();
```

Comment:
>var directiveTriviaCache = GetDirectiveTriviaCache(); [](http://example.com/codeflow?start=12&length=53)

The code looks fragile. Without deep knowledge of this component there is no way to tell if there is a chance that someone else is already using the list and we are about to clash. So,  I think it would be better to work directly with the field rather than with an innocent looking method. And I think we should set the field to null, while we are using the list.

Reply:
I like this suggestion, makes the calling code a bit more verbose, but more obvious about what's going on